### ISSUE TITLE
Navigation tree issue for non-staff users when DEFAULT_PERMISSION_CLASSES is customized

### ISSUE THREAD

**Issue by berekuk** [wagtail/wagtail#5695]
### Is your proposal related to a problem?

Wagtail admin displays "Server Error" in navigation tree in the following configuration:
* user doesn't have `is_staff` permission
* `DEFAULT_PERMISSION_CLASSES` for DRF is set to `rest_framework.permissions.IsAdminUser`

The error is caused by `{WAGTAIL_ROOT}/api/v2beta/pages/{id}` call, which is implemented in `admin.api.endpoints` endpoint, which doesn't have any permission classes set.

### Describe the solution you'd like

`admin.api.endpoints.PagesAdminAPIEndpoint` could check the same permission as Wagtail admin UI itself (I assume that's `Can access Wagtail admin` checkbox in groups settings).

### Describe alternatives you've considered

I could change one of my settings current settings but I don't want to do either of these:
* `is_staff` permission is for my team's staff, but I want non-staff users to submit changes for moderation.
* setting the restrictive `DEFAULT_PERMISSION_CLASSES` seems like a good practice (I don't want to miss `permission_classes` in some non-public view accidentally).

Another solution would be to implement `WAGTAIL_ADMIN_PERMISSION_CLASS` setting, but it seems like an overkill to me.

### Additional context

I can submit a PR if you're ok with the general idea of this issue, it doesn't seem that hard to fix :)

**Comment by gasman** [comment]
Hi @berekuk - are you testing against an up-to-date version of Wagtail? This sounds like a bug that was fixed in 2.6.3 and 2.7: https://github.com/wagtail/wagtail/pull/5643

**Comment by berekuk** [comment]
Yes, I'm on 2.7. The problem is not with `authentication_classes`, but with `permission_classes`.

To be clear, my DRF config includes this snippet, which causes the issue:

```
REST_FRAMEWORK = {
    ...
    'DEFAULT_PERMISSION_CLASSES': (
        'rest_framework.permissions.IsAdminUser',
    ),
    ...
}
```

When I comment it out, navigation tree works as expected.

**Comment by gasman** [comment]
Removing this from the 2.7.1 milestone - initially this looked like a variation of #5585, which was a regression in 2.6 caused by a change in permission logic (#5069). However, on further investigation, setting `DEFAULT_PERMISSION_CLASSES` also fails on older versions - so while this can still be considered a bug, it isn't one that's critical to fix on the 2.7 branch.

We'd be happy to accept a PR for this. Unfortunately it looks like it might not be entirely straightforward... I tried fixing it using the same method as #5643, but it appears that the `permission_classes` attribute is only respected if you set it on `BaseAPIEndpoint` (now `BaseAPIViewSet` in current master), not `PagesAdminAPIEndpoint` / `PagesAdminAPIViewSet` - and, of course, `BaseAPIEndpoint` is too general for this purpose.

**Comment by thibaudcolas** [comment]
This seems to be the same issue as #4083. Closing.
